packages/firestore/src/local/shared_client_state.ts

```diff
@@ -409,8 +538,27 @@ export class WebStorageSharedClientState implements SharedClientState {
     );
   }
 
+  private persistMutationState(
+    batchId: BatchId,
+    state: MutationBatchState,
+    error?: FirestoreError
+  ) {
+    const mutationState = new RemoteMutationBatch(
+      this.user,
+      batchId,
+      state,
+      error
+    );
+
+    const mutationKey = `${MUTATION_BATCH_KEY_PREFIX}_${this.persistenceKey}_${
+      this.user.uid
+    }_${batchId}`;
+
+    this.storage.setItem(mutationKey, mutationState.toLocalStorageJSON());
+  }
+
   /** Assembles the key for a client state in LocalStorage */
-  private toLocalStorageClientKey(clientKey: string): string {
+  private toLocalStorageClientStateKey(clientKey: string): string {
```

Comment:
Should the arg be `clientId: ClientKey` ?  This code is going to be so much less confusing when we change ClientKey to ClientId. :-)